src/components/SelectV2/Select.tsx

```diff
@@ -204,7 +203,7 @@ export const Select = ({
       >
         {variant !== 'default' && currentOption.startAdornment}{' '}
         {!variant.startsWith('short') &&
-          (currentOption.label || currentOption.value)}
+          (currentOption?.label || currentOption.value)}
```

Comment:
```suggestion
          (currentOption.label || currentOption.value)}
```

You don't need the optional chaining here, since currentOption is always an object.